src/Razor/src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.MvcApplicationPartsDiscovery.targets

```diff
@@ -25,12 +25,13 @@ Copyright (c) .NET Foundation. All rights reserved.
     </CoreCompileDependsOn>
 
     <_MvcApplicationPartAttributeGeneratedFile>$(IntermediateOutputPath)$(TargetName).MvcApplicationPartsAssemblyInfo$(DefaultLanguageSourceExtension)</_MvcApplicationPartAttributeGeneratedFile>
+    <_MvcApplicationPartCacheFile>$(IntermediateOutputPath)$(TargetName).MvcApplicationPartsAssemblyInfo.cache</_MvcApplicationPartCacheFile>
   </PropertyGroup>
 
   <Target
     Name="_DiscoverMvcApplicationParts"
     Inputs="$(ProjectAssetsFile);$(MSBuildAllProjects)"
-    Outputs="$(_MvcApplicationPartAttributeGeneratedFile)"
+    Outputs="$(_MvcApplicationPartCacheFile)"
```

Comment:
Deleting the file is a problem since it messes with incremental builds. Using a different file to indicate the current-ness of the discovery

Reply:
Ah because it was defined as an output.